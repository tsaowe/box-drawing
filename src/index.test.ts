import boxDraw from './index';

test('default tree', () => {
  const list = [
    {
      name: 'parent-folder',
      children: [
        {
          name: 'child-1',
          children: [
            {
              name: 'child-1-1',
            },
            {
              name: 'child-1-2',
            },
          ],
        },
        {
          name: 'child-2',
          children: [
            {
              name: 'child-2-1',
            },
            {
              name: 'child-2-2',
              children: [
                {
                  name: 'child-2-2-1',
                  children: [
                    {
                      name: 'child-2-2-1-1',
                    },
                    {
                      name: 'child-2-2-1-2',
                    },
                  ],
                },
              ],
            },
            {
              name: 'child-2-3',
            },
          ],
        },
      ],
    },
  ];

  expect(boxDraw(list)).toMatchSnapshot();
  expect(boxDraw(list, 2)).toMatchSnapshot();
});
