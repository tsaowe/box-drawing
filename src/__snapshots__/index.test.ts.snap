// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default tree 1`] = `
"└─ parent-folder 
    ├─ child-1 
    │    ├─ child-1-1 
    │    └─ child-1-2 
    └─ child-2 
        ├─ child-2-1 
        ├─ child-2-2 
        │    └─ child-2-2-1 
        │        ├─ child-2-2-1-1 
        │        └─ child-2-2-1-2 
        └─ child-2-3 "
`;

exports[`default tree 2`] = `
"└─ parent-folder -------------------
    ├─ child-1 ---------------------
    │    ├─ child-1-1 --------------
    │    └─ child-1-2 --------------
    └─ child-2 ---------------------
        ├─ child-2-1 ---------------
        ├─ child-2-2 ---------------
        │    └─ child-2-2-1 --------
        │        ├─ child-2-2-1-1 --
        │        └─ child-2-2-1-2 --
        └─ child-2-3 ---------------"
`;
